include/exadg/convection_diffusion/preconditioners/multigrid_preconditioner.cpp

```diff
@@ -250,25 +253,25 @@ MultigridPreconditioner<dim, Number>::initialize_operator(unsigned int const lev
 template<int dim, typename Number>
 void
 MultigridPreconditioner<dim, Number>::initialize_dof_handler_and_constraints(
-  bool const                         operator_is_singular,
-  dealii::FiniteElement<dim> const & fe,
-  Map_DBC const &                    dirichlet_bc,
-  Map_DBC_ComponentMask const &      dirichlet_bc_component_mask)
+  bool const                    operator_is_singular,
+  unsigned int const            n_components,
+  Map_DBC const &               dirichlet_bc,
+  Map_DBC_ComponentMask const & dirichlet_bc_component_mask)
 {
   Base::initialize_dof_handler_and_constraints(operator_is_singular,
-                                               fe,
+                                               n_components,
                                                dirichlet_bc,
                                                dirichlet_bc_component_mask);
 
   if(data.convective_problem and
      data.convective_kernel_data.velocity_type == TypeVelocityField::DoFVector)
   {
-    dealii::FESystem<dim> fe_velocity(dealii::FE_DGQ<dim>(fe.degree), dim);
+    dealii::FESystem<dim> fe_velocity(dealii::FE_DGQ<dim>(degree_velocity), dim);
```

Comment:
When looking again at the code, I think this line should be adjusted as well, e.g. in order to be prepared for simplex.